templates/types/streaming/fastapi/app/api/routers/vercel_response.py

```diff
@@ -53,17 +53,27 @@ async def content_generator(
         # Merge the chat response generator and the event generator
         combine = stream.merge(chat_response_generator, event_generator)
         is_stream_started = False
-        async with combine.stream() as streamer:
-            async for output in streamer:
-                if not is_stream_started:
-                    is_stream_started = True
-                    # Stream a blank message to start displaying the response in the UI
-                    yield cls.convert_text("")
-
-                yield output
-
-                if await request.is_disconnected():
-                    break
+        try:
+            async with combine.stream() as streamer:
+                async for output in streamer:
+                    if await request.is_disconnected():
+                        break
+
+                    if not is_stream_started:
+                        is_stream_started = True
+                        # Stream a blank message to start displaying the response in the UI
+                        yield cls.convert_text("")
+
+                    yield output
+        except Exception:
+            logger.exception("Error in stream response")
+            error_message = (
+                "An error occurred while processing your request. Please try again."
+            )
+            yield cls.convert_text(error_message)
+        finally:
+            # Ensure event handler is marked as done even if connection breaks
+            event_handler.is_done = True
```

Comment:
Why was this missing before?

Reply:
I can't remember the reason, but it looks to me like we had set the is_done flag before, but somehow forgot it when refactoring